clients/executor-http-client/src/lib.rs

```diff
@@ -13,10 +13,23 @@ pub enum Error {
     Request(reqwest::Error),
 }
 
+#[derive(Clone)]
+pub struct BasicAuthCredentials {
+    username: String,
+    password: Option<String>,
```

Comment:
Not sure it matters too much for this. But, is this safe? 
Is this related to the auth for the testnet? Does it ask the password in plaintext?